pkg/controller/nodeipam/ipam/multicidrset/multi_cidr_set_test.go

```diff
@@ -673,12 +673,24 @@ func TestCIDRSetv6(t *testing.T) {
 func TestMultiCIDRSetMetrics(t *testing.T) {
 	cidr := "10.0.0.0/16"
 	_, clusterCIDR, _ := utilnet.ParseCIDRSloppy(cidr)
+	clearMetrics(map[string]string{"clusterCIDR": cidr}, true)
+
 	// We have 256 free cidrs
 	a, err := NewMultiCIDRSet(clusterCIDR, 8)
 	if err != nil {
 		t.Fatalf("unexpected error creating MultiCIDRSet: %v", err)
 	}
-	clearMetrics(map[string]string{"clusterCIDR": cidr})
+
+	clusterMaskSize, _ := clusterCIDR.Mask.Size()
+	max := getMaxCIDRs(24, clusterMaskSize)
+	em := testMetrics{
+		usage:      0,
+		allocs:     0,
+		releases:   0,
+		allocTries: 0,
+		max:        float64(max),
```

Comment:
you have to set the expected max, if you use the value returned by the function the test will never fail if the function changes and returns an unexpected value

Reply:
i have no problem using the const int set to the expected value here, but how's using the return value of `getMaxCIDRs` invalid? since, this test is about whether the metrics are set as expected by the function called above and the same function is used there as well.

Reply:
scratch this comment

Reply:
yeah, I realized later sorry